src/Validator/PhoneNumberValidator.php

```diff
@@ -0,0 +1,81 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Validator;
+
+use EoneoPay\Externals\Validator\Interfaces\PhoneNumberValidatorInterface;
+use libphonenumber\NumberParseException;
+use libphonenumber\PhoneNumber;
+use libphonenumber\PhoneNumberFormat;
+use libphonenumber\PhoneNumberUtil;
+
+/**
+ * @SuppressWarnings(PHPMD.StaticAccess) Libphonenumber construction is with static methods
+ */
+class PhoneNumberValidator implements PhoneNumberValidatorInterface
+{
+    /**
+     * @var string
+     */
+    private $defaultRegion;
+
+    /**
+     * @var \libphonenumber\PhoneNumberUtil
+     */
+    private $parser;
+
+    /**
+     * Constructor
+     *
+     * @param string $defaultRegion
+     */
+    public function __construct(?string $defaultRegion = null)
+    {
+        $this->defaultRegion = $defaultRegion ?? 'AU';
+        $this->parser = PhoneNumberUtil::getInstance();
```

Comment:
Singleton ? I'm guessing this is something from the library?

Reply:
Yes

Reply:
🤐